torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -479,7 +462,10 @@ def init_unsharded_param(self):
             with torch.no_grad(), torch.autograd._unsafe_preserve_version_counter(
                 self._unsharded_param
             ):
-                torch.ops.fsdp.set_.default(self._unsharded_param, unsharded_param)
+                size = self._unsharded_param.numel() * self._unsharded_param.itemsize
+                if (storage := self._unsharded_param.untyped_storage()).size() != size:
+                    storage.resize_(size)
+                torch.ops.fsdp.copy_(self._unsharded_param, unsharded_param)
```

Comment:
hmm I still don't think I fully understand why the `copy_()` is necessary (I think.. we are already doing a `split_with_sizes_copy.out` that should be copying data from the allgather into the unsharded param somewhere?)

Reply:
chatted offline - `split_with_sizes_copy.out` will not directly write into `self._unsharded_param` storage. Instead, we rely on the `copy_()` here to copy the allgather output data into `self._unsharded_param`.

Reply:
another dumb question: do we end up seeing a `split_with_sizes_copy.out` op at all in the graph? (and if so what is the out= argument - do we have two fresh buffer intermediates in the graph, one corresponding to the ag_output, and another corresponding to the `split_with_sizes_copy` output?)

Reply:
> do we have two fresh buffer intermediates in the graph, one corresponding to the ag_output, and another corresponding to the split_with_sizes_copy output?

yes exactly - the graph looks like this: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmpigKLRV/rank_0/0_0_0/inductor_post_grad_graph_5.txt. `split_with_size_copy.out` is guaranteed not called on a graph input.